rmw_fastrtps_shared_cpp/test/test_names.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+
+#include "rmw/qos_profiles.h"
+
+#include "rmw_fastrtps_shared_cpp/names.hpp"
+#include "rmw_fastrtps_shared_cpp/namespace_prefix.hpp"
+
+TEST(NamespaceTest, get_prefix) {
+  EXPECT_EQ("", _get_ros_prefix_if_exists(""));
+  for (const auto & prefix : _get_all_ros_prefixes()) {
+    EXPECT_EQ(prefix, _get_ros_prefix_if_exists(prefix + "/test"));
+  }
+}
+
+TEST(NamespaceTest, strip_prefix) {
+  EXPECT_EQ("/no_prefix/test", _strip_ros_prefix_if_exists("/no_prefix/test"));
```

Comment:
I think strip_prefix and resolve_prefix should also test if its parameters are empty strings.

Reply:
Done in https://github.com/ros2/rmw_fastrtps/pull/388/commits/b029a312f4e049efbb609f1b9883650d7b5ffe4f. Had to change implementation a tiny bit for it to behave correctly in all cases.